general.el

```diff
@@ -647,90 +647,65 @@ globally (they have special interaction with other global keywords). :keymap, on
 
 ;; TODO better documentation
 (defun general-extended-def-:which-key (_state keymap key edef kargs)
-  "Add a which-key description for KEY.
+  " An alternative which-key implementation for General, using which-key's
+which-key-add-keymap-based-replacements.
+
+Add a which-key description for KEY.
 If :major-modes is specified in EDEF, add the description for the corresponding
 major mode. KEY should not be in the kbd format (kbd should have already been
 run on it)."
   (general-with-eval-after-load 'which-key
     (let* ((wk (general--getf2 edef :which-key :wk))
-           (major-modes (general--getf edef kargs :major-modes))
            (keymaps (plist-get kargs :keymaps))
-           ;; index of keymap in :keymaps
-           (keymap-index (cl-dotimes (ind (length keymaps))
-                           (when (eq (nth ind keymaps) keymap)
-                             (cl-return ind))))
-           (mode (let ((mode (if (and major-modes (listp major-modes))
-                                 (nth keymap-index major-modes)
-                               major-modes)))
-                   (if (eq mode t)
-                       (general--remove-map keymap)
-                     mode)))
            (key (key-description key))
-           (key-regexp (concat (when (general--getf edef kargs :wk-full-keys)
-                                 "\\`")
-                               (regexp-quote key)
-                               "\\'"))
            (prefix (plist-get kargs :prefix))
            (binding (or (when (and (plist-get edef :def)
                                    (not (plist-get edef :keymp)))
                           (plist-get edef :def))
                         (when (and prefix
                                    (string= key prefix))
                           (plist-get kargs :prefix-command))))
-           (replacement (cond ((stringp wk)
-                               (cons nil wk))
-                              (t
-                               wk)))
-           (match/replacement
-            (cons
-             (cons (when (general--getf edef kargs :wk-match-keys)
-                     key-regexp)
-                   (when (and (general--getf edef kargs :wk-match-binding)
-                              binding
-                              (symbolp binding))
-                     (regexp-quote (symbol-name binding))))
-             replacement)))
-      (general--add-which-key-replacement mode match/replacement)
-      (when (and (consp replacement)
-                 ;; lambda
-                 (not (functionp replacement)))
-        (general--add-which-key-title-prefix
-         mode key (cdr replacement))))))
+           (replacement (cond ((consp wk) (cdr wk))
+                              (t wk)))
+           ;; check the map exists, if not, try adding -map to it. (ie: global -> global-map)
+           (real-keymap (if (boundp keymap) (symbol-value keymap) (symbol-value (intern (format "%s-map" keymap)))))
```

Comment:
```suggestion
           (real-keymap (general--get-keymap nil keymap nil t)))
```
Have a look at `general--get-keymap`. I found it while browsing `general--define-key-dispatch`